src/controller/controller.ts

```diff
@@ -281,7 +283,9 @@ class Controller extends events.EventEmitter {
 
         clearInterval(this.backupTimer);
         clearInterval(this.databaseSaveTimer);
-        await this.backup();
+        try {
+            await this.backup();
+        } catch (e) {}
         await this.adapter.stop();
```

Comment:
In what situation does the backup fail but the adapter stop doesn't? I would expect the adapter.stop to also be in the try catch.

Reply:
Interestingly enough, the adapter.stop() does not throw an exception in any of the cases i was able to duplicate, while the this.backup still references uninitialised data structures. I could imagine placing it in the try/catch, but am hesitant to do so as i would want an error thrown on adapter.stop to be passed to the client application, while the error on backup would only be valid for cases where the backup should succeed.

Since the error messages are not currently parseable, this becomes impossible to determine, so i opted for catching the errors from being unable to perform a backup while keeping the errors on stop for the client application to handle.

Reply:
I managed to duplicate the backup failure tonight too. The failure originated in the test if use of backup was supported. On the zStack Adapter (the only one where i managed to recreate it), a crash was caused here:
```
    public async supportsBackup(): Promise<boolean> {
        return this.version.product !== ZnpVersion.zStack12;
    }
```
by `this.version` being `undefined`.

Due to other possible side effects i decided to keep this try/catch in the PR, but added an error message to be posted for information.

Reply:
What if you change `return this.version.product !== ZnpVersion.zStack12;` to `return this.version && this.version.product !== ZnpVersion.zStack12;`

Reply:
It will remove this one error, but i am hesitant to go that route, as it is just one of many errors i have seen which i am actually able to recreate. I am not convinced that i will not hit yet another error when this.version is populated but deeper in the backup functionality something is still uninitialized.

Reply:
@asgothian then we can fix that later, I think it's better than putting a try/catch for now.

Reply:
Ok.. i will run a test tomorrow - awaiting guests for tonight

Reply:
Updated. Was unable to test, as i could not get the error to reoccur. But i am sufficiently certain that it will work.